src/components/ui/Arc.tsx

```diff
@@ -0,0 +1,181 @@
+      currentY = keyPoints[2].y + (keyPoints[3].y - keyPoints[2].y) * easedProgress;
+    }
+
+    // 位置を更新
+    setOffset({
+      x: currentX,
+      y: currentY,
+    });
+
+    if (progress >= 1) {
+      // アニメーション終了
+      setTargeting(false);
+      return;
+    }
+
+    animationRef.current = requestAnimationFrame(animatePathMovement);
+  }, [targeting]);
+
+  // targetingがtrueになったときにパスアニメーションを開始する
+  useEffect(() => {
+    if (targeting) {
+      startTimeRef.current = Date.now();
+      // 初期位置を左斜め上に設定
+      setOffset({ x: -80, y: -60 });
+      animationRef.current = requestAnimationFrame(animatePathMovement);
+    }
+
+    return () => {
+      if (animationRef.current) {
+        cancelAnimationFrame(animationRef.current);
+      }
+    };
+  }, [targeting, animatePathMovement]);
+
+  // スキャンとターゲティングのサイクル
+  useEffect(() => {
+    // スキャンアニメーションを定期的に開始
+    const scanInterval = setInterval(() => {
+      // まずターゲティング（特定パスの動き）を開始
+      setTargeting(true);
+
+      // 2秒後にスキャンを開始
+      setTimeout(() => {
+        setScanning(true);
+        setTimeout(() => setScanning(false), 3000);
+      }, 2000);
+    }, 5000);
+
+    return () => {
+      clearInterval(scanInterval);
+    };
+  }, []);
+
+  return (
+    <Html position={position} rotation={rotation} transform distanceFactor={2} occlude={false}>
+      <div className='relative pointer-events-none'>
+        <div
+          className='relative scale-50 transform-gpu'
+          style={{
+            transform: `scale(0.5) translate(${offset.x}px, ${offset.y}px)`,
+            transition: targeting ? 'none' : 'transform 0.5s ease-out',
+          }}
+        >
+          {/* メインの円 */}
+          <div className={styles.circle}></div>
+
+          {/* 回転するアーク1 */}
+          <div className={styles.arcLayer1}></div>
+
+          {/* 回転するアーク2 */}
+          <div className={styles.arcLayer2}></div>
+
+          {/* 回転するアーク4 - 点線 */}
+          <div className={styles.arcDashed}></div>
+
+          {/* 回転するアーク5 - 部分的な弧 */}
+          <div className={styles.arcPartial}></div>
+
+          {/* ターゲティングレティクル */}
+          <div className={styles.targetingReticle}>
+            {/* 三角形マーカー - 円の外側に配置 */}
+            <div className={`${styles.triangleMarker} ${styles.topMarker}`}></div>
+            <div className={`${styles.triangleMarker} ${styles.rightMarker}`}></div>
+            <div className={`${styles.triangleMarker} ${styles.bottomMarker}`}></div>
+            <div className={`${styles.triangleMarker} ${styles.leftMarker}`}></div>
+          </div>
+
+          {/* スキャンエフェクト */}
+          <div className={`${styles.scanEffect} ${scanning ? styles.scanning : ''}`}></div>
+
+          {/* ターゲット状態表示 */}
+          <div className='absolute top-[-30px] left-1/2 transform -translate-x-1/2 text-xs font-mono text-green-500'>
+            {targeting ? 'TARGETING...' : scanning ? 'TARGET LOCKED' : 'STANDBY'}
+          </div>
```

Comment:
[IMO]
完全に好み & 老眼なだけかも何ですが、字ちっちゃくないですか？😂

ということで試しに以下を提案してみます！
※これはスルーでもOKです
```suggestion
          <div className='absolute top-[-30px] left-1/2 transform -translate-x-1/2 text-xl font-mono text-green-500'>
            {targeting ? 'TARGETING...' : scanning ? 'LOCKED' : 'STANDBY'}
          </div>
```